File: src/main/kotlin/com/example/kopring_api_server/domain/auth/config/SecurityConfig.java
package com.example.kopring_api_server.domain.auth.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

//@Configuration
//public class SecurityConfig {
//    @Bean
//    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
//        http
//                .formLogin(c -> c.disable())
//                .csrf(c -> c.disable())
//                .authorizeHttpRequests(c -> c
//                        .requestMatchers(
//                                "/**").permitAll()
//                        .anyRequest().authenticated())
//                .headers(c -> c.frameOptions(c2 -> c2.disable()));
//        return http.build();
//    }
//}

File: src/main/kotlin/com/example/kopring_api_server/domain/auth/config/SecurityConfig.kt
package com.example.kopring_api_server.domain.auth.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.SecurityBuilder
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.http.SessionCreationPolicy
import org.springframework.security.web.DefaultSecurityFilterChain

@Configuration
class SecurityConfig {
    @Bean
    fun filterChain(http: HttpSecurity): DefaultSecurityFilterChain = http
        .csrf { it.disable() }
        .authorizeHttpRequests {
            it.anyRequest().permitAll()
        }
        .sessionManagement { it.sessionCreationPolicy(SessionCreationPolicy.STATELESS) }
        .headers { it.frameOptions { it.disable() } }
        .build()
}